cmd/kubeadm/app/preflight/checks.go

```diff
@@ -243,6 +243,10 @@ func (hc HostnameCheck) Check() (warnings, errors []error) {
 	for _, msg := range validation.ValidateNodeName(hostname, false) {
 		errors = append(errors, fmt.Errorf("hostname \"%s\" %s", hostname, msg))
 	}
+	addr, err := net.LookupHost(hostname)
+	if addr == nil || err != nil {
```

Comment:
minor nit: if addr is nil here the error would be `hostname "{hostname}" ` which isn't very descriptive.
I think you can split that out to two cases.